### PR TITLE
e2e: fix authz invalid transfer authorisations test

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]

**Review by colin-axner — APPROVED** [review]

File: e2e/tests/transfer/authz_test.go [hunk 1]
```diff
@@ -248,7 +250,11 @@ func (suite *AuthzTransferTestSuite) TestAuthz_InvalidTransferAuthorizations() {
 			}
 
 			resp := suite.BroadcastMessages(context.TODO(), chainA, granteeWallet, msgExec)
-			suite.AssertTxFailure(resp, ibcerrors.ErrInsufficientFunds)
+			if testvalues.IbcErrorsFeatureReleases.IsSupported(chainAVersion) {
+				suite.AssertTxFailure(resp, ibcerrors.ErrInsufficientFunds)
+			} else {
+				suite.AssertTxFailure(resp, sdkerrors.ErrInsufficientFunds)
+			}
```
Comment (chatton) [inline e2e/tests/transfer/authz_test.go:257]: I'm a bit worried now about the sustainability of this approach, the tests will gradually get messier and messier and harder to work with with all these special cases.

I think there may be some work we can do in the future to come up with an alternate way of handling these changes.

**Review by chatton — APPROVED** [review]
LGTM! Thanks for fixing this. Left a comment but more a thought about possible improvements in the future.

File: e2e/tests/transfer/authz_test.go (hunk 1 above)
Comment (colin-axner) [reply e2e/tests/transfer/authz_test.go:257]: opened #3777

Comment (crodriguezvega) [reply e2e/tests/transfer/authz_test.go:257]: I agree. Let's open an issue and discuss in one of the engineering calls.

**Pull request merged**